jedi/evaluate/pep0484.py

```diff
@@ -102,7 +159,26 @@ def py__annotations__(funcdef):
 
 @evaluator_method_cache()
 def infer_return_types(function_context):
+    """
+    Infers the type of a function's return value,
+    according to type annotations.
+    """
     annotation = py__annotations__(function_context.tree_node).get("return", None)
+    if annotation is None:
+        # If there is no Python 3-type annotation, look for a Python 2-type annotation
+        node = function_context.tree_node
+        comment = parser_utils.get_following_comment_same_line(node)
+        if comment is None:
+            return NO_CONTEXTS
+
+        match = re.match(r"^#\s*type:\s*\([^#]*\)\s*->\s*([^#]*)", comment)
+        if not match:
+            return NO_CONTEXTS
+
+        annotation = tree.String(
+            repr(str(match.group(1).strip())),
+            node.start_pos)
+        annotation.parent = node.parent
```

Comment:
IMO we shoud not actually need to create a string object. We probably have to create a function similar to ` _evaluate_for_annotation` that can just take a string argument and use that to resolve comment annotations.

I know that it's already used once, so it's not your fault and you don't have to change it. I'll merge anyway if you don't do it in the next few days. If you don't do it I will :)